src/utils/chartJs/CustomTooltip.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+
+const styles = {
+  tooltip: {
+    position: 'absolute',
+    padding: '6px',
+    background: 'rgba(0, 0, 0, .8)',
+    color: 'white',
+    borderRadius: '4px',
+    pointerEvents: 'none',
+  },
+  tooltipKey: {
+    display: 'inline-block',
+    width: '10px',
+    height: '10px',
+    marginRight: '10px',
+  },
+};
+
+class CustomTooltip extends React.Component {
+  constructor(props) {
+    super(props);
+    this.tooltip = React.createRef();
+    this.state = { width: 0, height: 0 };
+  }
+
+  componentDidMount() {
+    this.setState({
+      width: this.tooltip.current.clientWidth,
+      height: this.tooltip.current.clientHeight,
+    });
+  }
+
+  render() {
+    const { classes, tooltipModel, series, canvas } = this.props;
+
+    if (tooltipModel.opacity === 0) return null;
+
+    let top = canvas.offsetTop + tooltipModel.caretY;
+    let left = canvas.offsetLeft + tooltipModel.caretX;
+
+    if (tooltipModel.yAlign === 'bottom') {
+      top -= this.state.height;
+    } else if (tooltipModel.yAlign === 'center') {
+      top -= 0.5 * this.state.height;
+    }
+
+    if (tooltipModel.xAlign === 'right') {
+      left -= this.state.width;
+    } else if (tooltipModel.xAlign === 'center') {
+      left -= 0.5 * this.state.width;
+    }
+
```

Comment:
I added new cases for vertical and horizontal alignment. Previously (in the other custom tooltip branch), it was only placed at the top or the bottom of the data point. Now it's placed in 3x2 positions all around the data point, which is the exact same implementation as the normal tooltips in ChartJS.